LondonDataServices.IDecide.Core/Services/Foundations/ConsumerAdoptions/ConsumerAdoptionService.cs

```diff
@@ -119,11 +119,13 @@ virtual internal async ValueTask BulkAddOrModifyBatchAsync(
                         .Select(consumerAdoption => new { consumerAdoption.DecisionId, consumerAdoption.ConsumerId })
                         .ToList();
 
-                    IQueryable<ConsumerAdoption> storageBatchConsumerAdoptions =
-                        (await this.storageBroker.SelectAllConsumerAdoptionsAsync())
-                            .Where(consumerAdoption => batchCompositeKeys.Any(
-                                key => key.DecisionId == consumerAdoption.DecisionId &&
-                                    key.ConsumerId == consumerAdoption.ConsumerId));
+                    // Materialize the query before using .Any() with a local collection
+                    var allConsumerAdoptions = (await this.storageBroker.SelectAllConsumerAdoptionsAsync()).ToList();
```

Comment:
You don't want to do this. It will materialize later down where you call `.ToList()`

What we had was correct

Reply:
You will need to fix the test.  The implementation is correct.